hardware_interface/include/hardware_interface/robot_hw.h

```diff
@@ -148,6 +148,25 @@ class RobotHW : public InterfaceManager
   virtual void doSwitch(const std::list<ControllerInfo>& /*start_list*/,
                         const std::list<ControllerInfo>& /*stop_list*/) {}
 
+  enum SwitchState
+  {
```

Comment:
Should an error code be also included with this or on the side, perhaps even as a string?

Reply:
I'd say that error reporting responsibility falls on the RobotHW implementation as this is hardware specific.

Reply:
Good point!